File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys
input = sys.stdin.readline

def solution():
  N = int(input())
  nums = [int(input()) for _ in range(N)]
  
  stack = []
  answer = []
  current = 1
  
  for num in nums:
    while current <= num:
      stack.append(current)
      answer.append('+')
      current += 1
      
    if stack[-1] == num:
      stack.pop()
      answer.append('-')
    else:
      print('NO')
      return
  
  print('\n'.join(answer))
    

solution()

File: 백준/Silver/1874. 스택 수열/스택 수열.js
function solution(input) {
  const N = parseInt(input[0], 10);
  const nums = input.slice(1, N + 1).map(Number);

  const stack = [];
  const answer = [];
  let current = 1;

  for (const num of nums) {
    while (current <= num) {
      stack.push(current);
      answer.push("+");
      current += 1;
    }

    if (stack[stack.length - 1] === num) {
      stack.pop();
      answer.push("-");
    } else {
      console.log("NO");
      return;
    }
  }

  console.log(answer.join("\n"));
}

const input = [];
require("readline")
  .createInterface({
    input: process.stdin,
    output: process.stdout,
  })
  .on("line", (line) => {
    input.push(line);
  })
  .on("close", () => {
    solution(input);
    process.exit();
  });
